src/studio/src/designer/backend/Services/Implementation/SourceControlSI.cs

```diff
@@ -553,6 +533,42 @@ public void VerifyCloneExists(string org, string repository)
             }
         }
 
+        private void CommitAndPushToBranch(string org, string repository, string branchName, string localPath, string message)
+        {
+            using (LibGit2Sharp.Repository repo = new LibGit2Sharp.Repository(localPath))
+            {
+                // Restrict users from empty commit
+                if (repo.RetrieveStatus().IsDirty)
+                {
+                    string remoteUrl = FindRemoteRepoLocation(org, repository);
+                    Remote remote = repo.Network.Remotes["origin"];
+
+                    if (!remote.PushUrl.Equals(remoteUrl))
+                    {
+                        // This is relevant when we switch beteen running designer in local or in docker. The remote URL changes.
+                        // Requires adminstrator access to update files.
+                        repo.Network.Remotes.Update("origin", r => r.Url = remoteUrl);
+                    }
+
+                    Branch b = repo.Branches[branchName];
+
+                    Commands.Stage(repo, "*");
+
+                    // Create the committer's signature and commit
+                    LibGit2Sharp.Signature author = new LibGit2Sharp.Signature(AuthenticationHelper.GetDeveloperUserName(_httpContextAccessor.HttpContext), "@jugglingnutcase", DateTime.Now);
```

Comment:
This could be refactored out into a small method to get the signature, the same lines of code is used two other places in this class.